dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -296,6 +409,9 @@ void WindowTransformAction::advanceFrameStart()
         break;
     }
     case WindowFrame::BoundaryType::Offset:
+        frame_start = stepForward(current_row, window_description.frame.begin_offset);
```

Comment:
The `stepForward` only works for `rows` frame type, need explicitly throw error if the frame type is range or groups.

Reply:
> The `stepForward` only works for `rows` frame type, need explicitly throw error if the frame type is range or groups.

Both `rows` and `range` will reach to here, so I think we should use the correct function according to frame type, instead of throwing exception.

Reply:
Does this means you will support `range` type in this pr?

Reply:
> Does this means you will support `range` type in this pr?

No, but I will throw exception for `range` type here.